src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -574,6 +592,12 @@ private void writeIdentity(final XMLStreamWriter streamWriter, final LoadedIdent
             streamWriter.writeEndDocument();
         }
 
+        public void dispose() {
+            // Release the write lock for this realm identity
+            final ReentrantReadWriteLock realmIdentityLock = getRealmIdentityLockForName(name);
+            realmIdentityLock.writeLock().unlock();
```

Comment:
Since this concept is shared by many realm implementations, a common realm IdentityLock is probably a good idea (could even reuse the Lock or ReadWriteLock interfaces for simplicity).  In terms of implementation, simple is fine: just use synchronization and wait/notify around a field or fields which implement the lock for now.  We can come up with something fancy later - if we need to!